iceoryx_hoofs/container/include/iox/detail/vector.inl

```diff
@@ -365,7 +408,10 @@ inline bool vector<T, Capacity>::erase(iterator position) noexcept
             at(n) = std::move(at(n + 1U));
             ++n;
         }
-        at(n).~T();
+        if constexpr (!std::is_trivial<T>::value)
+        {
+            at(n).~T();
+        }
         m_size--;
```

Comment:
For trivially movable types this might be optimized even further by using memmove to move the remaining elements instead of the while loop. I've no experience though to tell if it would be in fact faster.

Reply:
good catch, done!